src/acquisition/covidcast/csv_importer.py

```diff
@@ -282,17 +279,72 @@ def validate_missing_code(row, attr_quantity, attr_name, filepath=None, logger=N
 
     return missing_entry
 
+  @staticmethod
+  def geo_id_sanity_check(geo_type, geo_id):
+    """
+    Sanity check geo_id with respect to geo_type
+    """
+    def check_county():
+      if len(geo_id) != 5 or not '01000' <= geo_id <= '80000':
+        return (None, 'geo_id')
+      return geo_id
+    
+    def check_hrr():
+      if not 1 <= int(geo_id) <= 500:
+        return (None, 'geo_id')
+      return geo_id
+    
+    def check_msa():
+      if len(geo_id) != 5 or not '10000' <= geo_id <= '99999':
+        return (None, 'geo_id')
+      return geo_id
+    
+    def check_dma():
+      if not 450 <= int(geo_id) <= 950:
+        return (None, 'geo_id')
+      return geo_id
+
+    def check_state():
+      # note that geo_id is lowercase
+      if len(geo_id) != 2 or not 'aa' <= geo_id <= 'zz':
+        return (None, 'geo_id')
+      return geo_id
+
+    def check_hhs():
+      if not 1 <= int(geo_id) <= 10:
+        return (None, 'geo_id')
+      return geo_id
+
+    def check_nation():
+      # geo_id is lowercase
+      if len(geo_id) != 2 or not 'aa' <= geo_id <= 'zz':
+        return (None, 'geo_id')
+      return geo_id
+
+    geo_type_dict = {
+      'county': check_county(),
+      'hrr': check_hrr(),
+      'msa': check_msa(),
+      'dma': check_dma(),
+      'state': check_state(),
+      'hhs': check_hhs(),
+      'nation': check_nation(),
+    }
+
+    return geo_type_dict.get(geo_type, (None, 'geo_id'))
```

Comment:
```suggestion
    geo_type_dict = {
      'county': check_county,
      'hrr': check_hrr,
      'msa': check_msa,
      'dma': check_dma,
      'state': check_state,
      'hhs': check_hhs,
      'nation': check_nation,
    }

    return geo_type_dict.get(geo_type, lambda: (None, 'geo_id'))()
```
note: i did not test this, but hopefully it demonstrates the point